config/config.go

```diff
@@ -1371,6 +1378,31 @@ func migrateConfigDatabaseConnection(v *viper.Viper) {
 	}
 }
 
+// migrateConfigEventsEnabled is responsible for ensuring backward compatibility of events_enabled field.
+// This function copies the value of newField "events.enabled" and set it to the oldField "events_enabled".
+// This is necessary to achieve the desired order of precedence favoring the account values over the host values
+// given the account fetcher JSON merge mechanics.
+func migrateConfigEventsEnabled(oldFieldValue *bool, newFieldValue *bool) (updatedOldFieldValue, updatedNewFieldValue *bool) {
+	newField := "account_defaults.events.enabled"
+	oldField := "account_defaults.events_enabled"
+
+	if oldFieldValue != nil {
+		glog.Warningf("%s is deprecated and should be changed to %s", oldField, newField)
+	}
+	if newFieldValue != nil {
+		if oldFieldValue != nil {
+			glog.Warningf("using %s and ignoring deprecated %s", newField, oldField)
+		}
+	}
+
+	updatedOldFieldValue = oldFieldValue
+	if newFieldValue != nil {
```

Comment:
Niptpick: a small optimization would be to avoid the second `nil` check:
```
1385   func migrateConfigEventsEnabled(oldFieldValue *bool, newFieldValue *bool) (updatedOldFieldValue, updatedNewFieldValue *bool) {
1386       newField := "account_defaults.events.enabled"
1387       oldField := "account_defaults.events_enabled"
1388  
     +     updatedOldFieldValue = oldFieldValue
1389       if oldFieldValue != nil {
1390           glog.Warningf("%s is deprecated and should be changed to %s", oldField, newField)
1391       }
1392       if newFieldValue != nil {
1393           if oldFieldValue != nil {
1394               glog.Warningf("using %s and ignoring deprecated %s", newField, oldField)
1395           }
     +         updatedOldFieldValue = ptrutil.ToPtr(*newFieldValue)
1396       }
1397  
1398 -     updatedOldFieldValue = oldFieldValue
1399 -     if newFieldValue != nil {
1400 -         updatedOldFieldValue = ptrutil.ToPtr(*newFieldValue)
1401 -     }
1402  
1403       return updatedOldFieldValue, nil
1404   }
config/config.go
```